src/cal-heatmap.js

```diff
@@ -1868,7 +1871,7 @@ CalHeatMap.prototype = {
 		if (this.options.highlight.length > 0) {
 			for (var i in this.options.highlight) {
 				if (this.options.highlight[i] instanceof Date && this.dateIsEqual(this.options.highlight[i], d)) {
```

Comment:
You can remove the type check, since you added one in the `dateIsEqual` function